lms/djangoapps/course_api/blocks/serializers.py

```diff
@@ -77,7 +85,10 @@ def get_blocks(self, structure):
         """
         Serialize to a dictionary of blocks keyed by the block's usage_key.
         """
-        return {
-            unicode(block_key): BlockSerializer(block_key, context=self.context).data
-            for block_key in structure
-        }
+        result = {}
+        requested_block_types = None if len(self.context['block_types']) == 0 else self.context['block_types']
+        for block_key in structure:
+            block_type = structure.get_xblock_field(block_key, 'category')
+            if not requested_block_types or block_type in requested_block_types:
+                result[unicode(block_key)] = BlockSerializer(block_key, context=self.context).data
```

Comment:
This filtering shouldn't happen within the Serializer class.  Rather, please create a new Transformer for it (as noted in an earlier comment).